articles/javascript-higher-order-functions/index.md

```diff
@@ -0,0 +1,333 @@
+
+Functional programming is a significant building block for JavaScript as a language. They are some of the common scrips that you will continuously use when building JavaScript applications. Functions set statements to perform tasks. Functions define a task, which you can call and execute whenever you want inside your program.
+
+JavaScript treats a function as an object assigned to values to return a value. They have methods and properties, just like the other objects. Unlike ordinary objects, a function can be
+
+- Called.
+- Passed as an argument to another function.
+- Assigned to a variable as a value.
+- Be returned by another function.
+
+Functions are fundamental that outside the "normal" functions, JavaScript has high order functions.
+
+A high order function accepts another function as an argument or returns a function as its output. The concept is commonly referred to as a callback function. A function that takes another function as its argument is what we refer to as a callback function because it calls back that function.
+
+In this guide, we learn about the high order function. We will discuss and show you some of the common built-in JavaScript high order functions with examples. These include `filter`, `map`, `for each`, and `reduce`. To get along with this guide, prior knowledge to JavaScript [regular functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Functions) and [arrow functions](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions) will be important.
+
+### Array.prototype.filter
+
+`filter()` allows you to filter things in an array. It is a method that returns an array. It creates a new array of elements—the callback function test every input array element. The `filter()` method composes all the elements that pass the callback function test and return a new array.
+
+Let's digest this information with examples.
+
+Assume:
+1. An array with domestic animals such as
+
+```js
+["cat", "dog", "cow" ,"goat", "sheep", "donkey", "pig", "horse"]
+```
+
+We want to filter all domestic animals whose names have three-letter.
+
+2. An array of adult ages.
+
+```js
+[{name: "Xavier", age: 19},
+{name: "Alice", age: 21},
+{name: "John", age: 14},
+{name: "David", age: 22},
+{name: "Amanda", age: 18},
+{name: "Amy", age: 20},
+{name: "Jesicca", age: 28},
+{name: "Alex", age: 15}]
+```
+
+We want to filter adult ages 18 and older.
+
+#### Example One: without `filter()` Method
+
+##### Domestic Animals
+
+```js
+const Animals = ["cat", "dog", "cow" ,"goat", "sheep", "donkey", "pig", "horse"];
+const dAnimals = [];
+// we could use a standard for loop
+for (let i = 0; i < Animals.length; i++) {
+  if (Animals[i].length == 3) {
+    dAnimals.push(Animals[i])
+  }
+}
+Or a for of loop
```

Comment:
I think you mean for if loop. Also, display the output of the console.log statement. Include the console.log statement inside the for loop.

Reply:
Also, is it a comment? It's inside a code snippet.